content/blog/real-time-smart-city-traffic-monitoring-using-microservices-based-streaming-architecture-part-2.md

```diff
@@ -0,0 +1,331 @@
+
+The consumer side will have two tasks: to read the data from the stream and host an instance of a KieSession where the rule engine can apply rules on the facts as they are added to it.
+
+Rules are edited in the Workbench GUI, a Java webapp which can be run on a Java application server such as <a target='\_blank'  href='http://wildfly.org/'>WildFly</a> or <a target='\_blank'  href='http://tomcat.apache.org/'>Tomcat</a>.
+
+Rules are fetched from the workbench by the consumer application using the appropriate methods provided by the Drools framework, which are entirely based on Maven Repository.
+
+We can now look at the proposed system in more detail in the following sections.
+
+**List of Technologies Used**
+
+The technology we’re going to use is as follows:
+
+* MapR Sandbox 5.2
+* The Java programming language (any JVM language is fine)
+
+  * The Jackson 2 library to convert to and from JSON
+* MapR Event Store or Apache Kafka stream messaging system
+* Wildfly 10 application server to host the Workbench
+* <a target='\_blank'  href='https://www.drools.org/'>JBoss Drools</a> as our choice of OSS business rule engine
+* <a target='\_blank'  href='https://github.com/tdunning/log-synth'>Log Synth</a> to generate some synthetic data for our prototype
+* <a target='\_blank'  href='https://streamsets.com/'>Streamsets</a> 1.6 to connect MapR Event Store and Elasticsearch
+* Elasticsearch 2.4 and Kibana 4 for monitoring
+
+*Traffic Monitoring Prototype Architecture*
+
+![](/img/picture3.png)
+
+As I built this demo to run on the MapR Sandbox, I’m using instructions for CentOS 6.X, an open-source version of RHEL 6.X. Instructions for CentOS 7 are almost identical, and finding similar instructions for Ubuntu would be pretty straightforward and left up to the reader.
+
+To build the core of the traffic monitoring system, we’re going to need two basic parts:
+
+* A program to feed sensor data into MapR Event Store/Kafka. This part will be using fake data modeled by a vehicle simulation coded with Log Synth. We’ll use the MapR <a target='\_blank'  href='http://docs.confluent.io/2.0.0/kafka-rest/docs/index.html'>Kafka-rest</a> proxy implementation (just introduced with MEP 2.0)  to add the data with Python.
+* A JVM-language application that will read data from the stream and pass it to a KieSession. The minimal code to get this working is surprisingly small.
+
+To edit the rules, we deploy the Workbench on Wildfly 10, which is a fairly straightforward process. <a target='\_blank'  href='http://betzelblog.blogspot.jp/2015/02/setting-up-drools-workbench-and.html'>Check this blog post</a> for instructions, or read the Drools Documentation. Installing Wildfly is pretty simple; <a target='\_blank'  href='http://developer-should-know.com/post/112230363742/how-to-install-wildfly-as-a-service-on-linux'>see this blog</a> for great instructions on how to install it as a service on Centos/RHEL (it’s for Wildfly, but the same instructions work for 9 and 10).
```

Comment:
I clicked where you have the link "see this blog" and it came up with a dead link:

![image](https://user-images.githubusercontent.com/77496261/137515863-4ad9a1ec-ef4f-4d92-8c1c-b1e0628a8752.png)

Reply:
fixed